### PR TITLE
Added hyperlinks to icons and datasource labels

### LINKED ISSUE #489: Add links to data provider  for pathway Search and view
## Description of new feature

Alter existing text and icons in the PC search and view apps to provide hyperlinks to each data provider, when appropriate (i.e. a useful web page exists).  
image

image

## Motivation for new feature

Users should ideally, but not always, be able to link directly to an original data source. Source sites may provide additional information or greater detail about a pathway.

### PR DESCRIPTION
**Pull request by MichaelWrana**
Fixes #489 

Used already existing datasources.js file to associate each datasource with their respective website.

### CODE REVIEW

File: src/client/features/search/index.js [hunk 1]
```diff
@@ -11,6 +11,14 @@ const Icon = require('../../common/components').Icon;
 const { ServerAPI } = require('../../services');
 const Landing = require('./landing-box');
 
+//File in the scripts folder which has all the information needed to associate each URL with an ID
+//load that file and create a mapping object
+const URLData = require('../../../scripts/datasources/datasources');
```
Comment (d2fong) [inline src/client/features/search/index.js]: I am pretty sure the datasource info is already being loaded in the component.

See: https://github.com/PathwayCommons/app-ui/pull/745/files#diff-a1f9f779cabf8780553a2f0a90378e43L40

File: src/client/features/search/index.js [hunk 2]
```diff
@@ -125,16 +133,20 @@ class Search extends React.Component {
...
       });
+    
+    //get the link to source website from the map defined at start of file.
+    const sourceLink = sourceLinkMap[dsInfo.id] || "";
 
       return h('div.search-item', [
        h('div.search-item-icon',[
-          h('img', {src: dsInfo.iconUrl})
-        ]),
+          h('a',{href:sourceLink,target:"_blank"},[
```
Comment (d2fong) [inline src/client/features/search/index.js]: You should use the react router Link component.
https://github.com/PathwayCommons/app-ui/pull/745/files#diff-a1f9f779cabf8780553a2f0a90378e43L3
https://reacttraining.com/react-router/web/example/basic

File: src/client/features/search/index.js [hunk 3]
```diff
@@ -125,16 +133,20 @@ class Search extends React.Component {
...
+            h('img.search-item-image', {src: dsInfo.iconUrl})
+          ])]),
         h('div.search-item-content', [
           h(Link, { to: { pathname: '/view', search: queryString.stringify({ uri: result.uri }) }, target: '_blank' }, [
             h('h3.search-item-content-title', result.name || 'N/A'),
           ]),
-          h('p.search-item-content-datasource', ` ${dsInfo.name}`),
+          h('a.search-item-content-datasource',{href: sourceLink,target:"_blank"},`${dsInfo.name}`),
```
Comment (d2fong) [inline src/client/features/search/index.js]: Use the react router Link component

File: src/client/features/search/index.js (hunk 1 above)
Comment (MichaelWrana) [reply src/client/features/search/index.js]: I think the `dataSources` in the code you linked is coming from a file like:
http://apps.pathwaycommons.org/pc-client/datasources (live)
http://localhost:3000/pc-client/datasources (local version)
I'm not sure where that page is being created, and modifying the `datasources.js` file doesn't appear to change it.  It also doesn't contain the `homePage` property from the `datasources.js` file which I need.  Any advice?

**Pull request closed without merging**